ranger/ext/img_display.py

```diff
@@ -388,6 +416,52 @@ def _get_image_dimensions(path):
         return width, height
 
 
+@register_image_displayer("sixel")
+class SixelImageDisplayer(ImageDisplayer, FileManagerAware):
+    """Implementation of ImageDisplayer using SIXEL."""
+
+    def __init__(self):
+        self.win = None
+
+    def draw(self, path, start_x, start_y, width, height):
+        if self.win is None:
+            self.win = self.fm.ui.win.subwin(height, width, start_y, start_x)
+        else:
+            self.win.mvwin(start_y, start_x)
+            self.win.resize(height, width)
+
+        font_width, font_height = get_font_dimensions()
+        fit_width = font_width * width
+        fit_height = font_height * height
+
+        sixel_extra_args = shlex.split(self.fm.settings.sixel_extra_args)
+        result = check_output(["convert", path + "[0]",
+                               "-geometry", "{0}x{1}>"
+                               .format(fit_width, fit_height)]
+                              + sixel_extra_args
+                              + ["sixel:-"],
+                              stderr=PIPE)
+
+        with temporarily_moved_cursor(start_y, start_x):
+            if PY3:
```

Comment:
This now needs `from ranger import PY3`.

Reply:
What do you mean? It's already imported at line 34…

Reply:
Right, my error, it's in master. I was checking against stable. 🤦